nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/controller/parameter/StandardParameterProviderNode.java

```diff
@@ -0,0 +1,609 @@
+    public void verifyCanFetchParameters() {
+        final ValidationStatus validationStatus = performValidation();
+        if (validationStatus != ValidationStatus.VALID) {
+            throw new IllegalStateException(String.format("Parameter Provider [%s] cannot fetch parameters while validation state is %s",
+                    getIdentifier(), validationStatus));
+        }
+    }
+
+    @Override
+    public void fetchParameters() {
+        final ParameterProvider parameterProvider = parameterProviderRef.get().getParameterProvider();
+        final ConfigurationContext configurationContext = getConfigurationContext();
+        List<ParameterGroup> fetchedParameterGroups;
+        try (final NarCloseable narCloseable = NarCloseable.withComponentNarLoader(getExtensionManager(), parameterProvider.getClass(), parameterProvider.getIdentifier())) {
+            fetchedParameterGroups = parameterProvider.fetchParameters(configurationContext);
+        } catch (final IOException e) {
+            throw new RuntimeException(String.format("Error fetching parameters for Parameter Provider [%s]", getName()), e);
+        }
+
+        if (fetchedParameterGroups == null || fetchedParameterGroups.isEmpty()) {
+            return;
+        }
+
+        final Set<String> parameterGroupNames = new HashSet<>();
+
+        writeLock.lock();
+        try {
+            this.fetchedParameterGroups.clear();
+            for (final ParameterGroup group : fetchedParameterGroups) {
+                final String groupName = group.getGroupName();
+                if (parameterGroupNames.contains(groupName)) {
+                    throw new IllegalStateException(String.format("Parameter group [%s] is provided twice, which is not allowed", groupName));
+                }
+                final Collection<Parameter> parameters = group.getParameters();
+
+                if (parameters == null) {
+                    continue;
+                }
+                final List<Parameter> validParameters = new ArrayList<>();
+                final Set<String> parameterNames = new HashSet<>();
+                for (final Parameter parameter : parameters) {
+                    final ParameterDescriptor descriptor = parameter.getDescriptor();
+                    if (descriptor == null) {
+                        throw new IllegalStateException("All fetched parameters require a ParameterDescriptor");
+                    }
+                    final String parameterName = descriptor.getName();
+                    if (parameterNames.contains(parameterName)) {
+                        throw new IllegalStateException(String.format("Parameter [%s] is provided in group [%s] twice, which is not allowed",
+                                parameterName, groupName));
+                    }
+
+                    if (parameter.getValue() == null) {
+                        getLogger().warn("Skipping parameter [{}], which is missing a value", new Object[] {parameterName});
```

Comment:
```suggestion
                        getLogger().warn("Skipping parameter [{}], which is missing a value", parameterName);
```

Reply:
It looks like `TerminationAwareLogger`, which is what `getLogger()` returns, does not have a matching method signature.  I think the `Object[]` wrapper is required here.

Reply:
`TerminationAwareLogger` implements `ComponentLog`, which supports variable arguments, even though the implementation methods do not explicitly implement those signatures, so it should work unless I am missing something.